packages/gatsby-cli/src/reporter/reporter.ts

```diff
@@ -72,6 +113,24 @@ class Reporter {
     return process.exit(1)
   }
 
+  logFailureWithPluginName = ({
+    methodName,
+    pluginName,
+  }: {
+    methodName: string
+    pluginName: string
+  }) => (errorMeta: ErrorMeta, error?: Error | Array<Error>): void => {
+    if (typeof errorMeta === `object`) {
+      const id = errorMeta && errorMeta[`id`]
+
+      if (id) {
+        errorMeta[`id`] = `${pluginName}_${id}`
+      }
+    }
+
+    this[methodName](errorMeta, error)
+  }
```

Comment:
Same her do we want it on the reporter and not as a utility? I want to limit the public api from reporter.

Reply:
utility is good